modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

```diff
@@ -150,29 +150,52 @@ export class {{classname}} extends runtime.BaseAPI {
         {{/isOAuth}}
         {{/authMethods}}
         {{#hasFormParams}}
-        const formData = new FormData();
-        {{/hasFormParams}}
+        const consumes: runtime.Consume[] = [
+            {{#consumes}}
+            { contentType: '{{{mediaType}}}' },
+            {{/consumes}}
+        ];
+        // @ts-ignore: canConsumeForm may be unused
+        const canConsumeForm = runtime.canConsumeForm(consumes);
+
+        let formParams: { append(param: string, value: any): any };
+        let useForm = false;
+        {{#formParams}}
+        {{#isFile}}
+        // use FormData to transmit files using content-type "multipart/form-data"
+        useForm = canConsumeForm;
+        {{/isFile}}
+        {{/formParams}}
+        if (useForm) {
+            headerParameters['Content-Type'] = 'multipart/form-data';
```

Comment:
```suggestion
```
@tkrtmy wouldn't this override the `boundary` which would automatically be set? the `Content-Type` header is set automatically if `FormData` is passed as `body` parameter.
You can try it:
```javascript
// with FormData, withexplicit header
var formData = new FormData();
formData.append('something', 33);
fetch('https://jsonplaceholder.typicode.com/posts', {method: 'POST', headers:{'Content-Type': 'multipart/form-data'}, body: formData}).then(resp => console.log(resp))
// --> Invalid Request (Boundary missing):
// Content-Type: multipart/form-data

// with FormData, without explicit header
var formData = new FormData();
formData.append('something', 33);
fetch('https://jsonplaceholder.typicode.com/posts', {method: 'POST', body: formData}).then(resp => console.log(resp));
// --> Valid Request:
// Content-Type: multipart/form-data; boundary=----WebKitFormBoundarycA9KQvAXCq0NzmdA

// with URLSearchParams, without explicit header
var searchParams = new URLSearchParams();
formData.append('something', 33);
fetch('https://jsonplaceholder.typicode.com/posts', {method: 'POST', body: searchParams}).then(resp => console.log(resp))
// --> Valid Request:
// Content-Type: application/x-www-form-urlencoded;charset=UTF-8
```

Reply:
True! Thank you.
I'll fix it.